File: src/main/java/com/example/mapstruct/demo/EmployeeRepository.java
package com.example.mapstruct.demo;

import com.example.mapstruct.demo.entity.Employee;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.repository.CrudRepository;

public interface EmployeeRepository extends CrudRepository<Employee, Integer>, JpaSpecificationExecutor<Employee> {
}

File: src/main/kotlin/com/example/mapstruct/demo/EmployeeRepository.kt
package com.example.mapstruct.demo

import com.example.mapstruct.demo.entity.Employee
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface EmployeeRepository : CrudRepository<Employee?, Long?> {
    @Query("select e from Employee e where e.fio = :fio")
    fun findByFio(fio: String?): List<Employee?>?
}